sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -512,6 +512,9 @@ class Analyzer(
 
       // A special case for Generate, because the output of Generate should not be resolved by
       // ResolveReferences. Attributes in the output will be resolved by ResolveGenerate.
+      case g @ Generate(generator, _, _, _, _, _)
+        if !g.resolved && generator.resolved => g
+
       case g @ Generate(generator, join, outer, qualifier, output, child)
```

Comment:
I think this 2 cases can be simplified to:
```
case g: Generate if g.generator.resolved => g
case g @ Generate(generator, join, outer, qualifier, output, child) =>
  the generator resolution logic...
```

We only care about whether `generator` is resolved or not.